lib/activerecord-multi-tenant/query_rewriter.rb

```diff
@@ -324,6 +329,20 @@ def build_arel(*args)
 
       arel
     end
+
+    private
+    def relations_from_node_join(node_join)
+      if ActiveRecord::VERSION::MAJOR <= 5 || node_join.right.expr.is_a?(Arel::Nodes::Equality)
+        return node_join.right.expr.right.relation, node_join.right.expr.left.relation
+      end
+
+      children = node_join.right.expr.children
+      children.select{ |n| (n.is_a?(MultiTenant::TenantEnforcementClause) || n.is_a?(MultiTenant::TenantJoinEnforcementClause)) }.each do
+        return nil, nil
```

Comment:
I don't understand the return here - is our intention to always return on the first iteration of the loop?

Reply:
yes, so the thing is that I don't know if you have a way in ruby to say "if there is any objects of type `TenantEnforcementClause` or `TenantJoinEnforcementClause` return.

Reply:
Changed the condition here to include the case where a children is empty

Reply:
To further increase readability, I would move the `any?` into its own variable, and utilize the block syntax of `any?` (so we don't need the map)

```ruby
children = node_join.right.expr.children
tenant_applied = children.any? { n.is_a?(MultiTenant::TenantEnforcementClause) || n.is_a?(MultiTenant::TenantJoinEnforcementClause) }
return nil, nil if tenant_applied || children.empty?
```